datumaro/plugins/kitti_format/converter.py

```diff
@@ -0,0 +1,245 @@
+                            occluded = KittiPath.DEFAULT_OCCLUDED
+                            if 'truncated' in bbox.attributes and \
+                                self._allow_attributes:
+                                truncated = float(bbox.attributes['truncated'])
+                            if 'occluded' in bbox.attributes and \
+                                self._allow_attributes:
+                                occluded = int(bbox.attributes['occluded'])
+                            label_line[1] = truncated
+                            label_line[2] = occluded
+                            x, y, h, w = bbox.get_bbox()
+                            label_line[4:8] = x, y, x+h, y+w
+                            label_line = ' '.join([str(elem)
+                               for elem in label_line])
+                            f.writelines('%s\n' % label_line)
+
+        if KittiTask.segmentation in self._tasks:
+            self.save_label_map()
+
+    def get_label(self, label_id):
+        return self._extractor. \
+            categories()[AnnotationType.label].items[label_id].name
+
+    def save_label_map(self):
+        path = osp.join(self._save_dir, KittiPath.LABELMAP_FILE)
+        write_label_map(path, self._label_map)
+
+    def _load_categories(self, label_map_source):
+        if label_map_source == LabelmapType.kitti.name:
+            # use the default KITTI colormap
+            label_map = KittiLabelMap
+
+        elif label_map_source == LabelmapType.source.name and \
+                AnnotationType.mask not in self._extractor.categories():
+            # generate colormap for input labels
+            labels = self._extractor.categories() \
+                .get(AnnotationType.label, LabelCategories())
+            label_map = OrderedDict((item.name, None)
+                for item in labels.items)
+
+        elif label_map_source == LabelmapType.source.name and \
+                AnnotationType.mask in self._extractor.categories():
+            # use source colormap
+            labels = self._extractor.categories()[AnnotationType.label]
+            colors = self._extractor.categories()[AnnotationType.mask]
+            label_map = OrderedDict()
+            for idx, item in enumerate(labels.items):
+                color = colors.colormap.get(idx)
+                if color is not None:
+                    label_map[item.name] = color
+
+        elif isinstance(label_map_source, dict):
+            label_map = OrderedDict(
+                sorted(label_map_source.items(), key=lambda e: e[0]))
+
+        elif isinstance(label_map_source, str) and osp.isfile(label_map_source):
+            label_map = parse_label_map(label_map_source)
+
+        else:
+            raise Exception("Wrong labelmap specified, "
+                "expected one of %s or a file path" % \
+                ', '.join(t.name for t in LabelmapType))
+
+        self._categories = make_kitti_categories(label_map)
```

Comment:
Does this format require black color to have index 0?

Reply:
No